pulsar-client-cpp/.gitignore

```diff
@@ -77,6 +77,7 @@ Makefile
 cmake_install.cmake
 CMakeFiles
 CMakeCache.txt
+build/
```

Comment:
Do not commit unrelated ignore path here. A CMake build directory can be any name, for example, I prefer `_builds` directory in my own. Adding such directories to `.gitignore` is bad.